test/e2e/encryption_at_rest_test.go

```diff
@@ -90,20 +104,54 @@ var _ = Describe("Encryption at REST test", Label("encryption-at-rest"), func()
 				},
 			},
 		),
+		Entry("Test[encryption-at-rest-azure]: Can add Encryption at Rest to Azure project", Label("encryption-at-rest-azure"),
+			model.DataProvider(
+				"encryption-at-rest-azure",
+				model.NewEmptyAtlasKeyType().UseDefaultFullAccess(),
+				40000,
+				[]func(*model.TestDataProvider){},
+			).WithProject(data.DefaultProject()),
+			v1.EncryptionAtRest{
+				AzureKeyVault: v1.AzureKeyVault{
+					AzureEnvironment:  AzureEnvironment,
+					ClientID:          os.Getenv(AzureClientID),
+					Enabled:           toptr.MakePtr(true),
+					KeyVaultName:      KeyVaultName,
+					ResourceGroupName: cloud.ResourceGroupName,
+					Secret:            os.Getenv(AzureClientSecret),
+					TenantID:          os.Getenv(DirectoryID),
+					SubscriptionID:    os.Getenv(SubscriptionID),
+				},
+			},
+			nil,
+		),
+		Entry("Test[encryption-at-rest-gcp]: Can add Encryption at Rest to GCP project", Label("encryption-at-rest-gcp"),
+			model.DataProvider(
+				"encryption-at-rest-gcp",
+				model.NewEmptyAtlasKeyType().UseDefaultFullAccess(),
+				40000,
+				[]func(*model.TestDataProvider){},
+			).WithProject(data.DefaultProject()),
+			v1.EncryptionAtRest{
+				GoogleCloudKms: v1.GoogleCloudKms{
+					Enabled:           toptr.MakePtr(true),
+					ServiceAccountKey: os.Getenv("GCP_SA_CRED"),
+				},
+			},
+			nil,
+		),
 	)
 })
 
-func encryptionAtRestFlow(userData *model.TestDataProvider, encAtRest v1.EncryptionAtRest, roles []cloudaccess.Role) {
-	By("Add cloud access role (AWS only)", func() {
-		cloudAccessRolesFlow(userData, roles)
-	})
-
+func encryptionAtRestFlow(userData *model.TestDataProvider, encAtRest v1.EncryptionAtRest) {
 	By("Create KMS", func() {
 		Expect(userData.K8SClient.Get(userData.Context, types.NamespacedName{Name: userData.Project.Name,
 			Namespace: userData.Resources.Namespace}, userData.Project)).Should(Succeed())
 
-		Expect(len(userData.Project.Status.CloudProviderAccessRoles)).NotTo(Equal(0))
-		aRole := userData.Project.Status.CloudProviderAccessRoles[0]
+		var aRole status.CloudProviderAccessRole
+		if len(userData.Project.Status.CloudProviderAccessRoles) > 0 {
+			aRole = userData.Project.Status.CloudProviderAccessRoles[0]
+		}
```

Comment:
Should we not fail already here in the else case?

Reply:
I believe it is like this because this applies only to the AWS config. As the cloud provider configuration was happening before and now I moved it to even earlier, it is not supposed to fail ever.